libfoundation/src/foundation-foreign.cpp

```diff
@@ -322,39 +324,55 @@ __size_import (void *contents,
                bool release,
                MCValueRef & r_value)
 {
-	size_t t_value = *(size_t *) contents;
+    size_t t_value = *(size_t *) contents;
 
-	if (t_value > UINTEGER_MAX)
-	{
-		MCErrorCreateAndThrow (kMCForeignImportErrorTypeInfo,
-		                       "type", kMCSizeTypeInfo,
-		                       "reason", MCSTR("too large for Number representation"),
-		                       nil);
-		return false;
-	}
-
-	return MCNumberCreateWithUnsignedInteger((uinteger_t) t_value,
-	                                         (MCNumberRef &) r_value);
+    if (t_value <= UINTEGER_MAX)
+    {
+        return MCNumberCreateWithUnsignedInteger(t_value,
+                                                 (MCNumberRef &) r_value);
```

Comment:
Please use a C++ cast (e.g. `static_cast<MCNumberRef&>(r_value)`)

I know we were discussing adding a `MCValueCast` function, but for now, please avoid C-style casts, as agreed in the big meeting before Christmas!